test/controllers/deploys_controller_test.rb

```diff
@@ -1,7 +1,7 @@
 # frozen_string_literal: true
 require_relative '../test_helper'
 
-SingleCov.covered! uncovered: 1
+SingleCov.covered! uncovered: 2
```

Comment:
I incremented this because it was not straightforward to test the case where the deployer of another project can not cancel a deploy of this project. I assumed that would fall into the 'viewer' role, but it seems that role will be denied before it gets to the `can_be_cancelled_by` point.